projects/js-packages/connection/hooks/use-product-checkout-workflow/index.jsx

```diff
@@ -48,6 +51,18 @@ export default function useProductCheckoutWorkflow( {
 		isUserConnected
 	);
 
+	const handleAfterRegistration = () => {
+		return Promise.resolve( checkSiteHasWpcomProduct && checkSiteHasWpcomProduct() ).then(
+			siteHasWpcomProduct => {
+				if ( siteHasWpcomProduct ) {
+					handleConnectUser();
+				} else {
+					return ( window.location.href = checkoutProductUrl );
+				}
+			}
```

Comment:
we can simplify(?) this block.

```suggestion
				if ( siteHasWpcomProduct ) {
					return handleConnectUser();
				}

				window.location.href = checkoutProductUrl;
			}
```

I think the last `return` is not needed

Reply:
Good point. addressed in f688c4acb2